src/store/lib/useRekeyDateStore.js

```diff
@@ -37,26 +39,30 @@ export default function useRekeyDateStore() {
       const logs = await _contracts.azimuth.getPastEvents('ChangedKeys', {
         fromBlock: 0,
         toBlock: 'latest',
-        filter: { point: [point] },
+        filter: {
+          point: [point],
+        },
       });
 
       if (logs.length === 0) {
         //TODO: better encoding for "no rekeyDate" state?
         addToRekeyDateCache({
-          [point]: Just('unknown'),
+          [point]: Just(Result.Error('No rekey date available.')),
         });
       } else {
         // last log is most recent
         const blockNumber = logs[logs.length - 1].blockNumber;
         const block = await _web3.eth.getBlock(blockNumber);
         addToRekeyDateCache({
-          //TODO date format
-          [point]: Just(new Date(block.timestamp * 1000).toString()),
+          [point]: Just(Result.Ok(new Date(block.timestamp * 1000))),
```

Comment:
how do we feel about the `Maybe<Result<>>` pattern here? lets us handle the 'loading' state by checking for `Maybe.Nothing` and handle the failure state by checking `Maybe.Just(Result.Error)` and success by `Maybe.Just(Result.Ok)`